adagrams/game.py

```diff
@@ -1,11 +1,118 @@
+import random
+import string
 def draw_letters():
-    pass
+    hand_of_letters = []
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+    
+    #iterate through the dict to convert letter_pool to a list
+    LETTER_POOL_list = []  
+    for letter,count in LETTER_POOL.items():
+        while count > 0:
+            LETTER_POOL_list.append(letter)
+            count -=1    
+   
+    total_letter_count = len(LETTER_POOL_list)
+    hand_of_letters = []
+    #to get 10 letters 
+    for i in range(10):
+        #draw a random index from 0 to (totoal_letter_cout - i), given there are i letters
+        #there are already drawn & put back as the last i items in the list
+        draw_index = random.randrange(0, total_letter_count - i)
+        hand_of_letters.append(LETTER_POOL_list[draw_index])
+        # move the drawed letter to the end and won't be picked in next round
+        #swith two items in the list : a, b = b,a
+        LETTER_POOL_list[draw_index], LETTER_POOL_list[total_letter_count - i - 1] = \
+            LETTER_POOL_list[total_letter_count - i - 1], LETTER_POOL_list[draw_index]
+
+    return hand_of_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    `word`, describes some input word, and is a string
+    `letter_bank`, describes an array of drawn letters in a hand. 
+    It is an array of ten strings, with each string representing an uppercase letter
+    """
+    letter_bank_count= {}
+    for letter in letter_bank:
+        if letter not in letter_bank_count:
+            letter_bank_count[letter] = 1
+        else:
+            letter_bank_count[letter] += 1
+    
+    for char in word.upper():
+        if letter_bank_count.get(char,0) < 1:
```

Comment:
Missing white space after after the comma, it should look like `letter_bank_count.get(char, 0)`.

It's a small thing, but as you write more code (and eventually contribute to a large codebase) it's important to be consistent with whitespaces so that the whole project stays neat. Here's the official Python [style guide](https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements) that covers whitespaces.